MSSQL: dropAll() causes SQL Error [4104] [S0001]: The multi-part identifier "sys.schemas" could not be bound.

### Search first

- [X] I searched and no similar issues were found

### Description

We recently tried to upgrade from 4.9.1 to 4.23.1. Unfortunately, dropAll() fails in our development environment on MSSQL with the following error:

```
liquibase.exception.DatabaseException: Error executing SQL SELECT object_schema_name(object_id) AS schema_name, object_name(object_id) AS table_name, name AS column_name, CAST(seed_value AS bigint) AS start_value, CAST(increment_value AS bigint) AS increment_by FROM sys.identity_columns WHERE sys.schemas = 'dbo': The multi-part identifier "sys.schemas" could not be bound.
	at liquibase.executor.jvm.JdbcExecutor.execute(JdbcExecutor.java:98)
	at liquibase.executor.jvm.JdbcExecutor.query(JdbcExecutor.java:221)
	at liquibase.executor.jvm.JdbcExecutor.query(JdbcExecutor.java:229)
	at liquibase.executor.jvm.JdbcExecutor.queryForList(JdbcExecutor.java:294)
	at liquibase.executor.jvm.JdbcExecutor.queryForList(JdbcExecutor.java:289)
	at liquibase.snapshot.jvm.ColumnAutoIncrementService.obtainSequencesInformation(ColumnAutoIncrementService.java:55)
	at liquibase.snapshot.jvm.ColumnSnapshotGenerator.setAutoIncrementDetails(ColumnSnapshotGenerator.java:221)
	at liquibase.snapshot.jvm.ColumnSnapshotGenerator.addTo(ColumnSnapshotGenerator.java:203)
	at liquibase.snapshot.jvm.JdbcSnapshotGenerator.snapshot(JdbcSnapshotGenerator.java:78)
	at liquibase.snapshot.SnapshotGeneratorChain.snapshot(SnapshotGeneratorChain.java:49)
	at liquibase.snapshot.DatabaseSnapshot.include(DatabaseSnapshot.java:313)
	at liquibase.snapshot.DatabaseSnapshot.replaceObject(DatabaseSnapshot.java:456)
	at liquibase.snapshot.DatabaseSnapshot.replaceObject(DatabaseSnapshot.java:478)
	at liquibase.snapshot.DatabaseSnapshot.includeNestedObjects(DatabaseSnapshot.java:369)
	at liquibase.snapshot.DatabaseSnapshot.include(DatabaseSnapshot.java:332)
	at liquibase.snapshot.DatabaseSnapshot.init(DatabaseSnapshot.java:106)
	at liquibase.snapshot.DatabaseSnapshot.<init>(DatabaseSnapshot.java:59)
	at liquibase.snapshot.JdbcDatabaseSnapshot.<init>(JdbcDatabaseSnapshot.java:34)
	at liquibase.snapshot.SnapshotGeneratorFactory.createSnapshot(SnapshotGeneratorFactory.java:214)
	at liquibase.snapshot.SnapshotGeneratorFactory.createSnapshot(SnapshotGeneratorFactory.java:187)
	at liquibase.snapshot.SnapshotGeneratorFactory.createSnapshot(SnapshotGeneratorFactory.java:154)
	at liquibase.database.AbstractJdbcDatabase.dropDatabaseObjects(AbstractJdbcDatabase.java:797)
	at liquibase.command.core.DropAllCommandStep.run(DropAllCommandStep.java:65)
	at liquibase.command.CommandScope.execute(CommandScope.java:213)
	at liquibase.Liquibase.dropAll(Liquibase.java:952)
	at liquibase.Liquibase.dropAll(Liquibase.java:937)
	at com.example.migration.SpringLiquibase.initializeDatabase(SpringLiquibase.java:743)
	...
Caused by: com.microsoft.sqlserver.jdbc.SQLServerException: The multi-part identifier "sys.schemas" could not be bound.
	at com.microsoft.sqlserver.jdbc.SQLServerException.makeFromDatabaseError(SQLServerException.java:255)
	at com.microsoft.sqlserver.jdbc.SQLServerStatement.getNextResult(SQLServerStatement.java:1695)
	at com.microsoft.sqlserver.jdbc.SQLServerStatement.doExecuteStatement(SQLServerStatement.java:920)
	at com.microsoft.sqlserver.jdbc.SQLServerStatement$StmtExecCmd.doExecute(SQLServerStatement.java:814)
	at com.microsoft.sqlserver.jdbc.TDSCommand.execute(IOBuffer.java:7675)
	at com.microsoft.sqlserver.jdbc.SQLServerConnection.executeCommand(SQLServerConnection.java:4137)
	at com.microsoft.sqlserver.jdbc.SQLServerStatement.executeCommand(SQLServerStatement.java:272)
	at com.microsoft.sqlserver.jdbc.SQLServerStatement.executeStatement(SQLServerStatement.java:246)
	at com.microsoft.sqlserver.jdbc.SQLServerStatement.executeQuery(SQLServerStatement.java:736)
	at liquibase.executor.jvm.JdbcExecutor$QueryStatementCallback.doInStatement(JdbcExecutor.java:541)
	at liquibase.executor.jvm.JdbcExecutor.execute(JdbcExecutor.java:77)
	... 43 common frames omitted
```

Looking into this, it seems that a change introduced in #4140 and #4048 seems to cause this. In previous versions, the following query was generated:

```
SELECT
	object_schema_name(object_id) AS schema_name,
	object_name(object_id) AS table_name,
	name AS column_name,
	CAST(seed_value AS bigint) AS start_value,
	CAST(increment_value AS bigint) AS increment_by
FROM
	sys.identity_columns;
``` 
Starting with 4.23.0, this seem to be replaced with the following query, resulting in SQL Error [4104] [S0001]: The multi-part identifier "sys.schemas" could not be bound.

```
SELECT
	object_schema_name(object_id) AS schema_name,
	object_name(object_id) AS table_name,
	sys.identity_columns.name AS column_name,
	CAST(seed_value AS bigint) AS start_value,
	CAST(increment_value AS bigint) AS increment_by
FROM
	sys.identity_columns 
WHERE
	sys.schemas  = 'dbo';
``` 

Running the queries manually results in the same error. I have to admit I rarely work with MSSQL, therefore I cannot rule out any configuration issues entirely. 


### Steps To Reproduce

Liquibase is installed via Maven. We are using Spring Boot. The following code drops all from the database. This works well on other database systems, i.e., MariaDB and MySQL.

```java
SpringResourceAccessor resourceAccessor = new SpringResourceAccessor(resourceLoader);
try (Connection liquibaseConnection = getConnectionFromDriverManager(dataSourceProperties);
	Liquibase liquibase = new Liquibase(CHANGE_LOG, resourceAccessor, new JdbcConnection(liquibaseConnection));) {
	if (dropFirst) {
		liquibase.dropAll();
	}
}
``` 
The error does not occur when the database is empty. I would assume that ColumnSnapshotGenerator is not invoked (I have not debugged into this though). 

### Expected/Desired Behavior

All tables should be dropped from the database, allowing for a fresh initialization.

### Liquibase Version

5.23.1

### Database Vendor & Version

Microsoft SQL Server 2019

### Liquibase Integration

Maven, Spring Boot

### Liquibase Extensions

_No response_

### OS and/or Infrastructure Type/Provider

Windows Server 2022 Standard

### Additional Context

_No response_

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR (Thank you!)

Dear @tati-qalified ,

I stripped down our application code to a bare minimum, matching pretty much what you sent.


```java
@Service
public class DatabaseInitializationService implements ApplicationListener<ApplicationReadyEvent> {

	@Autowired
	private MySpringLiquibase mySpringLiquibase;

	@Autowired(required = false)
	private CommonSingleDbProperties commonSingleDbProperties;

	@Autowired
	private MyCommonProperties commonProperties;

	private void initializeOrMigrateSingleDatabase() throws LiquibaseException {
		MyDataSource dataSource = commonSingleDbProperties.getDatasource();
		DataSourceProperties dataSourceProperties = dataSource.asDataSourceProperties(commonProperties, false);
		mySpringLiquibase.initializeDatabase(dataSourceProperties, dataSource);
	}

	@Override
	public void onApplicationEvent(ApplicationReadyEvent event) {
		try {
			initializeOrMigrateSingleDatabase();
		} catch (Throwable e) {
			throw new RuntimeException(e);
		}
	}

}
``` 

```java
@Component
public class MySpringLiquibase implements ApplicationContextAware {

	public static ApplicationContext APPLICATION_CONTEXT;

	@Autowired
	private ResourceLoader resourceLoader;

	private Connection getConnectionFromDriverManager(DataSourceProperties dataSourceProperties) throws SQLException {
		String url = dataSourceProperties.getUrl();
		String user = dataSourceProperties.getUsername();
		String password = dataSourceProperties.getPassword();
		return DriverManager.getConnection(url, user, password);
	}

	@SuppressWarnings("deprecation")
	public void initializeDatabase(DataSourceProperties dataSourceProperties, MyDataSource dataSource) throws LiquibaseException {
		SpringResourceAccessor resourceAccessor = new SpringResourceAccessor(resourceLoader);
		try (Connection liquibaseConnection = getConnectionFromDriverManager(dataSourceProperties);
				Liquibase liquibase = new Liquibase("classpath:/META-INF/liquibase/liquibase-test.changelog.xml", resourceAccessor,
						new JdbcConnection(liquibaseConnection));) {
			liquibase.update();
			liquibase.dropAll();

		} catch (Exception e) {
			throw new LiquibaseException(e);
		}
	}

	@Override
	public void setApplicationContext(ApplicationContext applicationContext) throws BeansException {
		APPLICATION_CONTEXT = applicationContext;
	}

}
``` 

When running this, I get the following log. 

```

  .   ____          _            __ _ _
 /\\ / ___'_ __ _ _(_)_ __  __ _ \ \ \ \
( ( )\___ | '_ | '_| | '_ \/ _` | \ \ \ \
 \\/  ___)| |_)| | | | | || (_| |  ) ) ) )
  '  |____| .__|_| |_|_| |_\__, | / / / /
 =========|_|==============|___/=/_/_/_/
 :: Spring Boot ::                (v3.1.4)

2023-10-16T14:47:08.162+02:00  INFO 85412 --- [           main] com.example.deployer.app.DeployerApp     : Starting DeployerApp using Java 17.0.5 with PID 85412 (/home/jakob/workspace/liquibase_test/com.example.deployer.app/target/classes started by jakob in /home/jakob/workspace/liquibase_test/com.example.deployer.app)
2023-10-16T14:47:08.164+02:00  INFO 85412 --- [           main] com.example.deployer.app.DeployerApp     : The following 1 profile is active: "no-scope-cache"
2023-10-16T14:47:09.162+02:00  INFO 85412 --- [           main] a.c.p.web.common.config.MyDataSource   : Connecting to database jdbc:sqlserver://10.0.0.13:1433;databaseName=liquibase_evaluation;encrypt=false with username sa
2023-10-16T14:47:09.162+02:00  INFO 85412 --- [           main] a.c.p.web.common.config.MyDataSource   : Detected a SQL Server connection, setting org.quartz.impl.jdbcjobstore.MSSQLDelegate as quartz job store delegate
2023-10-16T14:47:10.445+02:00  INFO 85412 --- [           main] com.zaxxer.hikari.HikariDataSource       : HikariPool-1 - Starting...
2023-10-16T14:47:10.643+02:00  INFO 85412 --- [           main] com.zaxxer.hikari.pool.HikariPool        : HikariPool-1 - Added connection ConnectionID:1 ClientConnectionId: 85bf6a81-6d6b-4f2b-b959-38bb5b9c422f
2023-10-16T14:47:10.645+02:00  INFO 85412 --- [           main] com.zaxxer.hikari.HikariDataSource       : HikariPool-1 - Start completed.
2023-10-16T14:47:10.964+02:00  INFO 85412 --- [           main] org.quartz.impl.StdSchedulerFactory      : Using default implementation for ThreadExecutor
2023-10-16T14:47:10.976+02:00  INFO 85412 --- [           main] org.quartz.core.SchedulerSignalerImpl    : Initialized Scheduler Signaller of type: class org.quartz.core.SchedulerSignalerImpl
2023-10-16T14:47:10.977+02:00  INFO 85412 --- [           main] org.quartz.core.QuartzScheduler          : Quartz Scheduler v.2.3.2 created.
2023-10-16T14:47:10.984+02:00  INFO 85412 --- [           main] o.s.s.quartz.LocalDataSourceJobStore     : Detected usage of MSSQLDelegate class - defaulting 'selectWithLockSQL' to 'SELECT * FROM {0}LOCKS WITH (UPDLOCK,ROWLOCK) WHERE SCHED_NAME = {1} AND LOCK_NAME = ?'.
2023-10-16T14:47:10.985+02:00  INFO 85412 --- [           main] o.s.s.quartz.LocalDataSourceJobStore     : Using db table-based data access locking (synchronization).
2023-10-16T14:47:10.986+02:00  INFO 85412 --- [           main] o.s.s.quartz.LocalDataSourceJobStore     : JobStoreCMT initialized.
2023-10-16T14:47:10.986+02:00  INFO 85412 --- [           main] org.quartz.core.QuartzScheduler          : Scheduler meta-data: Quartz Scheduler (v2.3.2) 'mainScheduler' with instanceId 'NON_CLUSTERED'
  Scheduler class: 'org.quartz.core.QuartzScheduler' - running locally.
  NOT STARTED.
  Currently in standby mode.
  Number of jobs executed: 0
  Using thread pool 'org.quartz.simpl.SimpleThreadPool' - with 10 threads.
  Using job-store 'org.springframework.scheduling.quartz.LocalDataSourceJobStore' - which supports persistence. and is not clustered.

2023-10-16T14:47:10.986+02:00  INFO 85412 --- [           main] org.quartz.impl.StdSchedulerFactory      : Quartz scheduler 'mainScheduler' initialized from an externally provided properties instance.
2023-10-16T14:47:10.986+02:00  INFO 85412 --- [           main] org.quartz.impl.StdSchedulerFactory      : Quartz scheduler version: 2.3.2
2023-10-16T14:47:10.987+02:00  INFO 85412 --- [           main] org.quartz.core.QuartzScheduler          : JobFactory set to: org.springframework.scheduling.quartz.SpringBeanJobFactory@59c00010
2023-10-16T14:47:11.056+02:00  INFO 85412 --- [           main] com.example.deployer.app.DeployerApp     : Started DeployerApp in 3.239 seconds (process running for 3.692)
2023-10-16T14:47:11.067+02:00  INFO 85412 --- [           main] a.c.pro.web.common.AbstractService   : Starting Mail appender flush service
2023-10-16T14:47:11.067+02:00  INFO 85412 --- [           main] a.c.pro.web.common.AbstractService   : Started Mail appender flush service
2023-10-16T14:47:11.068+02:00  INFO 85412 --- [           main] a.c.p.web.common.config.MyDataSource   : Detected a SQL Server connection, setting org.quartz.impl.jdbcjobstore.MSSQLDelegate as quartz job store delegate
2023-10-16T14:47:11.211+02:00  INFO 85412 --- [           main] liquibase.database                       : Set default schema name to dbo
2023-10-16T14:47:11.497+02:00  INFO 85412 --- [           main] liquibase.changelog                      : Creating database history table with name: DATABASECHANGELOG
2023-10-16T14:47:11.529+02:00  INFO 85412 --- [           main] liquibase.changelog                      : Reading from DATABASECHANGELOG
2023-10-16T14:47:11.682+02:00  INFO 85412 --- [           main] liquibase.lockservice                    : Successfully acquired change log lock
2023-10-16T14:47:11.733+02:00  INFO 85412 --- [           main] liquibase.command                        : Using deploymentId: 7460431732
2023-10-16T14:47:11.737+02:00  INFO 85412 --- [           main] liquibase.changelog                      : Reading from DATABASECHANGELOG
Running Changeset: META-INF/liquibase/liquibase-test.changelog.xml::1::nvoxland
2023-10-16T14:47:11.767+02:00  INFO 85412 --- [           main] liquibase.changelog                      : Table person created
2023-10-16T14:47:11.769+02:00  INFO 85412 --- [           main] liquibase.changelog                      : ChangeSet META-INF/liquibase/liquibase-test.changelog.xml::1::nvoxland ran successfully in 11ms
Running Changeset: META-INF/liquibase/liquibase-test.changelog.xml::2::nvoxland
2023-10-16T14:47:11.793+02:00  INFO 85412 --- [           main] liquibase.changelog                      : Columns username(varchar(8)) added to person
2023-10-16T14:47:11.794+02:00  INFO 85412 --- [           main] liquibase.changelog                      : ChangeSet META-INF/liquibase/liquibase-test.changelog.xml::2::nvoxland ran successfully in 9ms
Running Changeset: META-INF/liquibase/liquibase-test.changelog.xml::3::nvoxland
2023-10-16T14:47:11.851+02:00  INFO 85412 --- [           main] liquibase.changelog                      : Lookup table added for person.state
2023-10-16T14:47:11.852+02:00  INFO 85412 --- [           main] liquibase.changelog                      : ChangeSet META-INF/liquibase/liquibase-test.changelog.xml::3::nvoxland ran successfully in 51ms
2023-10-16T14:47:11.861+02:00  INFO 85412 --- [           main] liquibase.util                           : UPDATE SUMMARY
2023-10-16T14:47:11.861+02:00  INFO 85412 --- [           main] liquibase.util                           : Run:                          3
2023-10-16T14:47:11.861+02:00  INFO 85412 --- [           main] liquibase.util                           : Previously run:               0
2023-10-16T14:47:11.861+02:00  INFO 85412 --- [           main] liquibase.util                           : Filtered out:                 0
2023-10-16T14:47:11.862+02:00  INFO 85412 --- [           main] liquibase.util                           : -------------------------------
2023-10-16T14:47:11.862+02:00  INFO 85412 --- [           main] liquibase.util                           : Total change sets:            3


UPDATE SUMMARY
Run:                          3
Previously run:               0
Filtered out:                 0
-------------------------------
Total change sets:            3

2023-10-16T14:47:11.862+02:00  INFO 85412 --- [           main] liquibase.util                           : Update summary generated
2023-10-16T14:47:11.863+02:00  INFO 85412 --- [           main] liquibase.command                        : Update command completed successfully.
Liquibase: Update has been successful. Rows affected: 3
2023-10-16T14:47:11.872+02:00  INFO 85412 --- [           main] liquibase.lockservice                    : Successfully released change log lock
2023-10-16T14:47:11.873+02:00  INFO 85412 --- [           main] liquibase.command                        : Command execution complete
2023-10-16T14:47:11.898+02:00  INFO 85412 --- [           main] liquibase.lockservice                    : Successfully acquired change log lock
2023-10-16T14:47:11.899+02:00  INFO 85412 --- [           main] liquibase.command                        : Dropping Database Objects in schema: liquibase_evaluation.dbo
2023-10-16T14:47:12.046+02:00  INFO 85412 --- [           main] liquibase.snapshot                       : Could not read identity information

liquibase.exception.DatabaseException: Error executing SQL SELECT object_schema_name(object_id) AS schema_name, object_name(object_id) AS table_name, name AS column_name, CAST(seed_value AS bigint) AS start_value, CAST(increment_value AS bigint) AS increment_by FROM sys.identity_columns WHERE sys.schemas = 'dbo': The multi-part identifier "sys.schemas" could not be bound.
	at liquibase.executor.jvm.JdbcExecutor.execute(JdbcExecutor.java:98)
	at liquibase.executor.jvm.JdbcExecutor.query(JdbcExecutor.java:221)
	at liquibase.executor.jvm.JdbcExecutor.query(JdbcExecutor.java:229)
	at liquibase.executor.jvm.JdbcExecutor.queryForList(JdbcExecutor.java:294)
	at liquibase.executor.jvm.JdbcExecutor.queryForList(JdbcExecutor.java:289)
	at liquibase.snapshot.jvm.ColumnAutoIncrementService.obtainSequencesInformation(ColumnAutoIncrementService.java:55)
	at liquibase.snapshot.jvm.ColumnSnapshotGenerator.setAutoIncrementDetails(ColumnSnapshotGenerator.java:221)
	at liquibase.snapshot.jvm.ColumnSnapshotGenerator.addTo(ColumnSnapshotGenerator.java:203)
	at liquibase.snapshot.jvm.JdbcSnapshotGenerator.snapshot(JdbcSnapshotGenerator.java:78)
	at liquibase.snapshot.SnapshotGeneratorChain.snapshot(SnapshotGeneratorChain.java:49)
	at liquibase.snapshot.DatabaseSnapshot.include(DatabaseSnapshot.java:313)
	at liquibase.snapshot.DatabaseSnapshot.replaceObject(DatabaseSnapshot.java:456)
	at liquibase.snapshot.DatabaseSnapshot.replaceObject(DatabaseSnapshot.java:478)
	at liquibase.snapshot.DatabaseSnapshot.includeNestedObjects(DatabaseSnapshot.java:369)
	at liquibase.snapshot.DatabaseSnapshot.include(DatabaseSnapshot.java:332)
	at liquibase.snapshot.DatabaseSnapshot.init(DatabaseSnapshot.java:106)
	at liquibase.snapshot.DatabaseSnapshot.<init>(DatabaseSnapshot.java:59)
	at liquibase.snapshot.JdbcDatabaseSnapshot.<init>(JdbcDatabaseSnapshot.java:34)
	at liquibase.snapshot.SnapshotGeneratorFactory.createSnapshot(SnapshotGeneratorFactory.java:214)
	at liquibase.snapshot.SnapshotGeneratorFactory.createSnapshot(SnapshotGeneratorFactory.java:187)
	at liquibase.snapshot.SnapshotGeneratorFactory.createSnapshot(SnapshotGeneratorFactory.java:154)
	at liquibase.database.AbstractJdbcDatabase.dropDatabaseObjects(AbstractJdbcDatabase.java:797)
	at liquibase.command.core.DropAllCommandStep.run(DropAllCommandStep.java:65)
	at liquibase.command.CommandScope.execute(CommandScope.java:213)
	at liquibase.Liquibase.dropAll(Liquibase.java:952)
	at liquibase.Liquibase.dropAll(Liquibase.java:937)
	at com.exmaple.migration.MySpringLiquibase.initializeDatabase(MySpringLiquibase.java:43)
	at com.example.deployer.app.DatabaseInitializationService.initializeOrMigrateSingleDatabase(DatabaseInitializationService.java:30)
	at com.example.deployer.app.DatabaseInitializationService.onApplicationEvent(DatabaseInitializationService.java:36)
	at com.example.deployer.app.DatabaseInitializationService.onApplicationEvent(DatabaseInitializationService.java:1)
	at org.springframework.context.event.SimpleApplicationEventMulticaster.doInvokeListener(SimpleApplicationEventMulticaster.java:174)
	at org.springframework.context.event.SimpleApplicationEventMulticaster.invokeListener(SimpleApplicationEventMulticaster.java:167)
	at org.springframework.context.event.SimpleApplicationEventMulticaster.multicastEvent(SimpleApplicationEventMulticaster.java:145)
	at org.springframework.context.support.AbstractApplicationContext.publishEvent(AbstractApplicationContext.java:437)
	at org.springframework.context.support.AbstractApplicationContext.publishEvent(AbstractApplicationContext.java:370)
	at org.springframework.boot.context.event.EventPublishingRunListener.ready(EventPublishingRunListener.java:109)
	at org.springframework.boot.SpringApplicationRunListeners.lambda$ready$6(SpringApplicationRunListeners.java:80)
	at java.base/java.lang.Iterable.forEach(Iterable.java:75)
	at org.springframework.boot.SpringApplicationRunListeners.doWithListeners(SpringApplicationRunListeners.java:118)
	at org.springframework.boot.SpringApplicationRunListeners.doWithListeners(SpringApplicationRunListeners.java:112)
	at org.springframework.boot.SpringApplicationRunListeners.ready(SpringApplicationRunListeners.java:80)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:334)
	at com.example.deployer.app.DeployerApp.runWithInitializer(DeployerApp.java:62)
	at com.example.deployer.app.DeployerApp.main(DeployerApp.java:38)
Caused by: com.microsoft.sqlserver.jdbc.SQLServerException: The multi-part identifier "sys.schemas" could not be bound.
	at com.microsoft.sqlserver.jdbc.SQLServerException.makeFromDatabaseError(SQLServerException.java:255)
	at com.microsoft.sqlserver.jdbc.SQLServerStatement.getNextResult(SQLServerStatement.java:1695)
	at com.microsoft.sqlserver.jdbc.SQLServerStatement.doExecuteStatement(SQLServerStatement.java:920)
	at com.microsoft.sqlserver.jdbc.SQLServerStatement$StmtExecCmd.doExecute(SQLServerStatement.java:814)
	at com.microsoft.sqlserver.jdbc.TDSCommand.execute(IOBuffer.java:7675)
	at com.microsoft.sqlserver.jdbc.SQLServerConnection.executeCommand(SQLServerConnection.java:4137)
	at com.microsoft.sqlserver.jdbc.SQLServerStatement.executeCommand(SQLServerStatement.java:272)
	at com.microsoft.sqlserver.jdbc.SQLServerStatement.executeStatement(SQLServerStatement.java:246)
	at com.microsoft.sqlserver.jdbc.SQLServerStatement.executeQuery(SQLServerStatement.java:736)
	at liquibase.executor.jvm.JdbcExecutor$QueryStatementCallback.doInStatement(JdbcExecutor.java:541)
	at liquibase.executor.jvm.JdbcExecutor.execute(JdbcExecutor.java:77)
	... 43 common frames omitted

2023-10-16T14:47:12.515+02:00  INFO 85412 --- [           main] liquibase.database                       : Successfully deleted all supported object types in schema liquibase_evaluation.dbo.
2023-10-16T14:47:12.519+02:00  WARN 85412 --- [           main] liquibase.lockservice                    : Failed to release change log lock
All objects dropped from sa@jdbc:sqlserver://10.0.0.13:1433;connectRetryInterval=10;connectRetryCount=1;maxResultBuffer=-1;sendTemporalDataTypesAsStringForBulkCopy=true;delayLoadingLobs=true;useFmtOnly=false;useBulkCopyForBatchInsert=false;cancelQueryTimeout=-1;sslProtocol=TLS;jaasConfigurationName=SQLJDBCDriver;statementPoolingCacheSize=0;serverPreparedStatementDiscardThreshold=10;enablePrepareOnFirstPreparedStatementCall=false;fips=false;socketTimeout=0;authentication=NotSpecified;authenticationScheme=nativeAuthentication;xopenStates=false;datetimeParameterType=datetime2;sendTimeAsDatetime=true;replication=false;trustStoreType=JKS;trustServerCertificate=false;TransparentNetworkIPResolution=true;iPAddressPreference=IPv4First;serverNameAsACE=false;sendStringParametersAsUnicode=true;selectMethod=direct;responseBuffering=adaptive;queryTimeout=-1;packetSize=8000;multiSubnetFailover=false;loginTimeout=30;lockTimeout=-1;lastUpdateCount=true;prepareMethod=prepexec;encrypt=false;disableStatementPooling=true;databaseName=liquibase_evaluation;columnEncryptionSetting=Disabled;applicationName=Microsoft JDBC Driver for SQL Server;applicationIntent=readwrite;
2023-10-16T14:47:12.524+02:00  INFO 85412 --- [           main] liquibase.command                        : Command execution complete
2023-10-16T14:47:12.539+02:00  INFO 85412 --- [           main] o.s.s.quartz.SchedulerFactoryBean        : Shutting down Quartz Scheduler
2023-10-16T14:47:12.540+02:00  INFO 85412 --- [           main] org.quartz.core.QuartzScheduler          : Scheduler mainScheduler_$_NON_CLUSTERED shutting down.
2023-10-16T14:47:12.540+02:00  INFO 85412 --- [           main] org.quartz.core.QuartzScheduler          : Scheduler mainScheduler_$_NON_CLUSTERED paused.
2023-10-16T14:47:12.540+02:00  INFO 85412 --- [           main] org.quartz.core.QuartzScheduler          : Scheduler mainScheduler_$_NON_CLUSTERED shutdown complete.
2023-10-16T14:47:12.541+02:00  INFO 85412 --- [           main] a.c.pro.web.common.AbstractService   : Stopping Mail appender flush service
2023-10-16T14:47:12.541+02:00  INFO 85412 --- [           main] a.c.pro.web.common.AbstractService   : Stopped Mail appender flush service
2023-10-16T14:47:12.548+02:00  INFO 85412 --- [           main] com.zaxxer.hikari.HikariDataSource       : HikariPool-1 - Shutdown initiated...
2023-10-16T14:47:12.549+02:00  INFO 85412 --- [           main] com.zaxxer.hikari.HikariDataSource       : HikariPool-1 - Shutdown completed.
``` 

The stacktrace is still in the log, it seems to be logged as INFO though and the database seems to get dropped afterwards. I hope this helps to reproduce the issue. If I can be of any help, please let me know. 

Best,
Jakob

--- a/liquibase-standard/src/main/java/liquibase/snapshot/jvm/ColumnAutoIncrementService.java
+++ b/liquibase-standard/src/main/java/liquibase/snapshot/jvm/ColumnAutoIncrementService.java
@@ -81,7 +81,7 @@ public class ColumnAutoIncrementService {
                     "CAST(seed_value AS bigint) AS start_value, " +
                     "CAST(increment_value AS bigint) AS increment_by " +
                     "FROM sys.identity_columns " +
-                    "WHERE sys.schemas = '" + schema.getName() + "'";
+                    "WHERE object_schema_name(object_id) = '" + schema.getName() + "'";
         } else if (database instanceof PostgresDatabase) {
             int version = 9;
             try {

